util/util.go

```diff
@@ -130,3 +132,18 @@ func CoordinatorCertChainFromPEM(pemChain []byte) (rootCert, intermediateCert *x
 
 	return rootCert, intermediateCert, nil
 }
+
+// AddOEQuoteHeader adds an OpenEnclave quote header to the given quote.
+// If the quote already has a header, this is a no-op.
+func AddOEQuoteHeader(quote []byte) []byte {
+	quoteHeader := make([]byte, 16)
+	binary.LittleEndian.PutUint32(quoteHeader, 1)     // version
+	binary.LittleEndian.PutUint32(quoteHeader[4:], 2) // OE_REPORT_TYPE_SGX_REMOTE
+	binary.LittleEndian.PutUint64(quoteHeader[8:], uint64(len(quote)))
+
+	// If the quote already has a header, return it as is.
+	if len(quote) > 8 && slices.Equal(quoteHeader[:8], quote[:8]) {
```

Comment:
OE has at least two supported headers, so this will break if it's the other one. Checking whether the quote is a raw SGX quote seems to be the more correct way. However, SGX quotes also have multiple versions. So we can either check for the version known to us. Or we just try to verify the quote as is and if it fails, we try again with added header. A bit ugly, but probably stable also for future changes to OE or SGX quote versions.